### PR TITLE
proved a few kernelc list examples

### PR DESCRIPTION
**Pull request by yilongli**

### CODE REVIEW

**Comment by yilongli** [comment]
Please don't review yet. I want to rebase this pull request on https://github.com/kframework/k/pull/860/.

**Comment by yilongli** [comment]
@andreistefanescu @dwightguth please review. Since I have made quite some change to the code itself, I would like to get it merged asap. I will revise the spec files to use the new notation in the next pull request.

File: samples/kernelc/tests/list/add_spec.k [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                (next |-> struct listNode *)
+                (val  |-> int)
+            )
+        </struct>
+        <fun>
+            add |-> functionBody ( int v, struct listNode * x , struct listNode *
+              , {  { struct listNode * y ; y = (( struct listNode * ) (malloc ( (
+              sizeof ( struct listNode )) ))) ; (y -> val) = v ; (y -> next) = x ;
+               return y ; } return tv ( void , undef ) ; } )
+        //...</fun> // TODO(YilongL): had to disable `...` to forbid the prover
```
Comment (andreistefanescu) [inline samples/kernelc/tests/list/add_spec.k]: I fixed this by changing the kernelc semantics.

**Comment by yilongli** [comment]
Actually, I just revised all the specification files. The task turns out to be quite simple.

File: samples/kernelc/tests/list/add_spec.k (hunk 1 above)
Comment (yilongli) [inline samples/kernelc/tests/list/add_spec.k]: yeah, I know

File: samples/kernelc/tests/list/add_spec.k [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                    // applying function call rule on `malloc` operation which
+                    // generates UnsuppportedOperationException's and leads to
+                    // illegal states which can not be proved
+        </fun>
+        <k>
+            //add(v, x) => tv(struct listNode *, object(?Ly:Int))
+            add(v, x) => tv(struct listNode *, ?Y:Pointer)
```
Comment (andreistefanescu) [inline samples/kernelc/tests/list/add_spec.k]: I would prefer `add(tv(int, V:Int), tv(struct listNode *, P:Pointer))`. Both are correct, but the specification above cannot be applied on add(1 + 1, ...).

Comment (yilongli) [inline samples/kernelc/tests/list/add_spec.k]: Andrei, I just pushed a new commit that revised all the spec files. You are reviewing the outdated diff.

File: samples/kernelc/tests/list/length_recursive_spec.k [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2014 K Team. All Rights Reserved.
+require "../../kernelc.k"
+
+module LENGTH-RECURSIVE-SPEC
+  imports KERNELC
+
+rule [loop-inv]:
```
Comment (andreistefanescu) [inline samples/kernelc/tests/list/length_recursive_spec.k]: This rule is not needed, just the one below.

**Comment by andreistefanescu** [comment]
Great job Yilong. You can merge.

**Pull request merged**